### PR TITLE
Handle race by treating cluster as disconnected.

### PR DESCRIPTION
**Pull request by cmeiklejohn**
Prevent a hard crash when the cluster has been connected, but the
gen_server has not been registered because we're still in the process of
reconfiguring after a topology change in riak_repl2_pg_block_requester.

### CODE REVIEW

File: src/riak_repl_pb_get.erl [hunk 1]
```diff
@@ -170,7 +170,15 @@ proxy_get_13(State, CName, CNames, B, K, GetOptions) ->
...
                     Leader = riak_core_cluster_mgr:get_leader(),
                     ProxyForCluster = riak_repl_util:make_pg_proxy_name(ClusterName),
-                    gen_server:call({ProxyForCluster, Leader}, {proxy_get, B, K, GetOptions}, ?LONG_TIMEOUT);
+                    try gen_server:call({ProxyForCluster, Leader},
+                                        {proxy_get, B, K, GetOptions},
+                                        ?LONG_TIMEOUT) of
+                        Result ->
+                            Result
+                    catch
+                        _:_ ->
```
Comment (bookshelfdave) [inline src/riak_repl_pb_get.erl]: seems like a lager:debug() might be useful here

**Comment by bookshelfdave** [comment]
all existing riak_test:replication2_pg tests are passing.

**Comment by cmeiklejohn** [comment]
Amended commit with logging, look good?

**Comment by bookshelfdave** [comment]
+1

**Comment by cmeiklejohn** [comment]
Do we also want to merge this to 2.0?

**Comment by bookshelfdave** [comment]
yes please

**Pull request merged**